procrastinate/sql/schema.sql

```diff
@@ -86,30 +86,52 @@ CREATE INDEX procrastinate_periodic_defers_job_id_fkey_v1 ON procrastinate_perio
 CREATE INDEX idx_procrastinate_workers_last_heartbeat ON procrastinate_workers(last_heartbeat);
 
 -- Functions
-CREATE FUNCTION procrastinate_defer_job_v1(
-    queue_name character varying,
-    task_name character varying,
-    priority integer,
-    lock text,
-    queueing_lock text,
-    args jsonb,
-    scheduled_at timestamp with time zone
+CREATE FUNCTION procrastinate_defer_jobs_v1(
+    queue_names character varying[],
+    task_names character varying[],
+    priorities integer[],
+    locks text[],
+    queueing_locks text[],
+    args_list jsonb[],
+    scheduled_ats timestamp with time zone[]
 )
```

Comment:
I've never playd with that, but I believe you can create your own PG composite types, so instead of arrays of values (where you wish all arrays have the same length), you could define a single array of composite values

https://www.postgresql.org/docs/current/rowtypes.html

Reply:
Yes, good idea. Unfortunately, it is not so easy to realise as the composite types would have to be registered in the connectors, see [psycopg doc](https://www.psycopg.org/psycopg3/docs/basic/pgtypes.html). I am a bit hesitant to create such a dependency between the connectors and the schema (especially as the current implementation works just fine). What do you think?

Reply:
I was wrong ... registering component types only seems to be relevant when returning a result. I'll try again using component types.

Reply:
Ok, finally got it working 😁.